### ISSUE TITLE
Users should be able to specify their desired topology semantics

### ISSUE THREAD

**Issue by billonahill** [twitter/heron#1904]
Instead of having to configure `setAckingEnabled(boolean)` and/or `setTopologyExactlyOnceEnabled(boolean)` of the Heron config objects, users should be able to set their tuple semantics as one of a mutually exclusive set of supported options. We could do this with something like this:

```
enum TupleDeliverySemantics {
  AT_LEAST_ONCE,
  AT_MOST_ONCE,
  EXACTLY_ONCE
}

Config {
  setTupleDeliverySemantics(TupleDeliverySemantics semantics);
}
```

This will allow users to set configs in the way that we discuss semantics. The `setTopologyExactlyOnceEnabled()` can be removed since we haven't shipped that yet and the  `setEnableAcking(..)` method should be deprecated from user-facing config APIs once we support setting the processing semantics. See additional discussion in #1898.

We should do this before shipping support for `setTopologyExactlyOnceEnabled()` so we don't need to break that API later.

**Comment by maosongfu** [comment]
👍

**Comment by srkukarni** [comment]
Solved by #2089
